lib/action_controller/responder.rb

```diff
@@ -129,7 +129,8 @@ class Responder
     def initialize(controller, resources, options={})
       @controller = controller
       @request = @controller.request
-      @format = @controller.formats.first
+      
+      @format = @controller.formats.detect{|f| Renderers::RENDERERS.include?(f) } || @controller.formats.first
```

Comment:
Just to be clear, what I mean is: If you have a controller like:

```
class FoosController < AC::Base
  respond_to :omg, :json
```

and the accept header has `text/omg` but there is no renderer registered for `:omg` responders should still use the `:omg` mime type in the response.